Content.Server/Defusable/Systems/DefusableSystem.cs

```diff
@@ -58,38 +118,68 @@ public void TryDetonateBomb(EntityUid uid, DefusableComponent comp)
         if (!comp.BombLive)
             return;
 
-        Logger.Debug("boom");
-        _explosionSystem.TriggerExplosive(uid);
+        _popup.PopupEntity(Loc.GetString("defusable-popup-boom", ("name", uid)), uid);
+
+        _explosion.TriggerExplosive(uid);
+        RemComp<ActiveDefusableComponent>(uid);
         QueueDel(uid);
 
         UpdateAppearance(uid, comp);
     }
 
-    public void DefuseBomb(EntityUid uid, DefusableComponent comp)
+    public void TryDefuseBomb(EntityUid uid, DefusableComponent comp)
     {
-        // todo: defusing lmfao
-        // also might want to have admin logs
+        if (!comp.BombLive)
+            return;
 
-        Logger.Debug("counter terrorists win");
+        _popup.PopupEntity(Loc.GetString("defusable-popup-defuse", ("name", uid)), uid);
         comp.BombLive = false;
         comp.BombUsable = false; // fry the circuitry
+        RemComp<ActiveDefusableComponent>(uid);
 
         UpdateAppearance(uid, comp);
     }
 
-    public void Update()
+    public override void Update(float frameTime)
     {
-        // todo: handle bombs
+        base.Update(frameTime);
+
+        foreach (var activeComp in EntityQuery<ActiveDefusableComponent>())
```

Comment:
use EntityQueryEnumerator and while (query.MoveNext(out var uid, out var activeComp))

Reply:
irrelevant now